### PR TITLE
 framework/st_things: Improve st_things stack robustness

### PR DESCRIPTION
**Pull request by zhouxinhe**
- Change a stack object `is_wifi_retry_connect` to be static
- Update logic to make ping thread desrtoyed promptly

### CODE REVIEW

**Comment by zhouxinhe** [comment]
@Hyunjun85 Could you please help to review?

**Comment by seinfra** [comment]
Target : [8b2c4f1713af3b996a4789d521c4268d37cb4bdd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8b2c4f1713af3b996a4789d521c4268d37cb4bdd] - Code Rule Check OK.

File: framework/src/st_things/things_stack/easy-setup/easysetup_manager.c [hunk 1]
```diff
@@ -456,7 +456,7 @@ static void *wifi_prov_set_loop(void *param)
 	THINGS_LOG_D(TAG, "Open Loop Enter.");
 
 	static int timeout_handle_num = 100;
-	things_timeout_s timeout;
+	static things_timeout_s timeout;
```
Comment (ChinaTizenRT) [inline framework/src/st_things/things_stack/easy-setup/easysetup_manager.c]: could give more concrete info?  timeout is stack var,  who will free it beforeused in new threads as param?

Comment (zhouxinhe) [reply framework/src/st_things/things_stack/easy-setup/easysetup_manager.c]: The reference of this stack object `timeout` is passed to new thread at line 495, 
and all stack objects will be freed after current function returned at line 501. 
In the new thread `timeout` dereference may happen later, but `timeout` stack object
does not exist any more at that time.

**Comment by seinfra** [comment]
Target : [1d048ba096044004de91033fb3e878bd2d0c6b63] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1d048ba096044004de91033fb3e878bd2d0c6b63] - Code Rule Check OK.

File: framework/src/st_things/things_stack/easy-setup/easysetup_manager.c (hunk 1 above)
Comment (zhouxinhe) [reply framework/src/st_things/things_stack/easy-setup/easysetup_manager.c]: As disscussed offline, this object will be cloned when creating new thread, there's no problem!
I reverted the change, thanks for your review!

**Pull request merged**